MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.mm

```diff
@@ -420,8 +421,20 @@
 	getContent(_mtlPrimitiveTopology, isDynamic) = mvkMTLPrimitiveTypeFromVkPrimitiveTopology(topology);
 }
 
+// Metal does not support VK_POLYGON_MODE_POINT, but it can be emulated if the polygon mode
+// is static, which allows both the topology and the pipeline topology-class to be set to points.
+// This cannot be accomplished if the dynamic polygon mode has been changed to points when the
+// pipeline is expecting triangles or lines, because the pipeline topology class will be incorrect.
 MTLPrimitiveType MVKRenderingCommandEncoderState::getPrimitiveType() {
-	return getMTLContent(PrimitiveTopology);
+	if (isDynamicState(PolygonMode) &&
+		_isPolygonModePoint[StateScope::Dynamic] &&
+		!_isPolygonModePoint[StateScope::Static]) {
+		
+		reportWarning(VK_ERROR_FEATURE_NOT_PRESENT, "vkCmdSetPolygonMode(): Metal does not support setting VK_POLYGON_MODE_POINT dynamically.");
+		return getMTLContent(PrimitiveTopology);
+	}
+
+	return getContent(_isPolygonModePoint, PolygonMode) ? MTLPrimitiveTypePoint : getMTLContent(PrimitiveTopology);
```

Comment:
If we can't actually set it to `VK_POLYGON_MODE_POINT` dynamically... can we even claim to support `extendedDynamicState3PolygonMode`?

Reply:
Wireframe is still a valid (IMHO more so than points) use of polygon mode, so supporting it is still useful.

Making a distinction was the purpose of the `VkPhysicalDevicePortabilitySubsetFeaturesKHR::pointPolygons` (in final to be named `rasterizationPolygonModePoints`). With the static emulation in place, we can probably now limit this to `rasterizationPolygonModePointsDynamic`

Reply:
Very well, then.